zipline/pipeline/mixins.py

```diff
@@ -109,7 +111,16 @@ def _compute(self, windows, dates, assets, mask):
         compute = self.compute
         missing_value = self.missing_value
         params = self.params
-        out = full_like(mask, missing_value, dtype=self.dtype)
+        outputs = self.outputs
+        if outputs:
+            out = recarray(
+                mask.shape,
+                formats=[self.dtype.str] * len(outputs),
```

Comment:
We might at some point want to support different dtypes for the different fields.  It's probably a fine restriction to start with that multi-output factors have to have a fixed dtype for all fields, but we should call that out somewhere in the docs.